### PR TITLE
Make DDP Quantization hooks backend Agnostic

### PR DESCRIPTION
**Pull request by VRSinghHabana**
Current ddp hooks quantization code use .cuda() API to move tensors and parameter on backend devices. This limits only cuda backend to work with ddp quantization hooks.
Change is to make code backend agnostic and move tensors/parameters based on **tensor.device.**

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by fegin — COMMENTED** [review]
LGTM, I'll wait for the CI signals.

**Comment by VRSinghHabana** [comment]
@fegin , CI is passed, Could you pls merge the change.

File: torch/distributed/algorithms/ddp_comm_hooks/quantization_hooks.py [hunk 1]
```diff
@@ -4,27 +4,27 @@
 from torch import nn
 
 
-def _quantize_per_tensor_cuda(x, scale, zero_point):
+def _quantize_per_tensor_backend(x, scale, zero_point):
```
Comment (kwen2501) [inline torch/distributed/algorithms/ddp_comm_hooks/quantization_hooks.py:7]: nit: maybe just name it `_quantize_per_tensor`?

File: torch/distributed/algorithms/ddp_comm_hooks/quantization_hooks.py [hunk 2]
```diff
@@ -4,27 +4,27 @@
...
 
 
-def _dequantize_per_tensor_cuda(y, scale, zero_point):
+def _dequantize_per_tensor_backend(y, scale, zero_point):
     x = scale * (y.to(torch.float32) - zero_point)
     return x
 
 
-def _quantize_per_channel_cuda(x, scale, zero_point):
+def _quantize_per_channel_backend(x, scale, zero_point):
```
Comment (kwen2501) [inline torch/distributed/algorithms/ddp_comm_hooks/quantization_hooks.py:18]: nit: maybe just name it `_quantize_per_channel`?

**Review by kwen2501 — APPROVED** [review]
LGTM

**Pull request closed without merging**